common/lib/xmodule/xmodule/x_module.py

```diff
@@ -398,7 +398,7 @@ def get_content_titles(self):
         the display_names of all its leaf nodes.
         """
         if self.has_children:
-            return sum((child.get_content_titles() for child in self.get_children()), [])
+            return sum((child.get_content_titles() for child in self.get_display_items()), [])
```

Comment:
The intention is that `get_content_titles` in particular xblocks that only display subsets of their children (like `SplitTestModule`) would do the relevant checks. The reason that it's been working for `SplitTestModule` up until now is that that XModule owns the knowledge of which child to show to a user. The group access decision, on the other hand, is handled by the `has_access` call at the time that children are loaded, which happens in `get_children`.

We should make sure that `get_children` is always triggering the `has_access` check (which it should be, as long as there is a `ModuleSystem` in place).